salt/config/__init__.py

```diff
@@ -2848,14 +2848,14 @@ def is_profile_configured(opts, provider, profile_name, vm_=None):
     # If cloning on Linode, size and image are not necessary.
     # They are obtained from the to-be-cloned VM.
     if driver == 'linode' and profile_key.get('clonefrom', False):
-        non_image_drivers.append('linode')
+        drivers_requiring_an_image.append('linode')
         non_size_drivers.append('linode')
 
     # If cloning on VMware, specifying image is not necessary.
-    if driver == 'vmware' and profile_key.get('image', True):
-        non_image_drivers.append('vmware')
+    if driver == 'vmware' and profile_key.get('clonefrom') is None:
+        drivers_requiring_an_image.append('vmware')
 
-    if driver not in non_image_drivers:
+    if driver in drivers_requiring_an_image:
```

Comment:
That's completely incorrect. The change you made would mean that drivers like ``nova``, ``virtualbox`` require ``image`` to be specified which is completely the opposite. No need to change this line at all.